superset-frontend/src/datasource/ChangeDatasourceModal.tsx

```diff
@@ -47,86 +68,93 @@ const TABLE_COLUMNS = [
   const [filter, setFilter] = useState<any>(undefined);
-  const [loading, setLoading] = useState(true);
+  const [confirmChange, setConfirmChange] = useState(false);
+  const [confirmedDataset, setConfirmedDataset] = useState<any>(undefined);
   let searchRef = useRef<HTMLInputElement>(null);
 
-  useEffect(() => {
-    const selectDatasource = (datasource: any) => {
-      SupersetClient.get({
-        endpoint: `/datasource/get/${datasource.type}/${datasource.id}`,
-      })
-        .then(({ json }) => {
-          onDatasourceSave(json);
-          onChange(datasource.uid);
-        })
-        .catch(response => {
-          getClientErrorObject(response).then(
-            ({ error, message }: { error: any; message: string }) => {
-              const errorMessage = error
-                ? error.error || error.statusText || error
-                : message;
-              addDangerToast(errorMessage);
-            },
-          );
-        });
-      onHide();
-    };
+  const {
+    state: { loading, resourceCollection },
+    fetchData,
+  } = useListViewResource<Dataset>('dataset', t('dataset'), addDangerToast);
+
+  const selectDatasource = useCallback(
+    (datasource: { type: string; id: number; uid: string }) => {
+      setConfirmChange(true);
+      setConfirmedDataset(datasource);
+    },
+    [],
+  );
 
-    const onEnterModal = () => {
+  useDebouncedEffect(() => {
+    fetchData({
+      pageIndex: 0,
+      pageSize: 20,
+      filters: [
+        {
+          id: 'table_name',
+          operator: 'ct',
+          value: filter,
+        },
+      ],
+      sortBy: [{ id: 'changed_on_delta_humanized' }],
+    });
+  }, 1000);
+
+  useEffect(() => {
+    const onEnterModal = async () => {
       if (searchRef && searchRef.current) {
         searchRef.current.focus();
       }
-      if (!datasources) {
-        SupersetClient.get({
-          endpoint: '/superset/datasources/',
-        })
-          .then(({ json }) => {
-            const data = json.map((ds: any) => ({
-              rawName: ds.name,
-              connection: ds.connection,
-              schema: ds.schema,
-              name: (
-                <a
-                  href="#"
-                  onClick={() => selectDatasource(ds)}
-                  className="datasource-link"
-                >
-                  {ds.name}
-                </a>
-              ),
-              type: ds.type,
-            }));
-            setLoading(false);
-            setDatasources(data);
-          })
-          .catch(response => {
-            setLoading(false);
-            getClientErrorObject(response).then(({ error }: any) => {
-              addDangerToast(error.error || error.statusText || error);
-            });
-          });
-      }
+
+      // Fetch initial datasets for tableview
+      await fetchData({
+        pageIndex: 0,
+        pageSize: 20,
+        filters: [],
+        sortBy: [{ id: 'changed_on_delta_humanized' }],
+      });
```

Comment:
Nit: you might want to move this to a const, eg:

```javascript
const emptyRequest = {
  pageIndex: 0,
  pageSize: 20,
  filters: [],
  sortBy: [{ id: 'change_on_delta_humanized' }],
};
```

Then in your `useDebounceEffect` you can reuse it too:

```javascript
fetchData({
  ...emptyRequest,
  filters: [{ id: 'table_name', 'operator': 'ct', value: 'filter' }],
});
```